### DISCUSSION TITLE
DataGrid Popup force keep open

### DISCUSSION THREAD

**Discussion by tomironkko** [Megabit/Blazorise#2046] [Q&A] [answered]
Hi,

Is it possible to force the Popup to keep it open in DataGrid Edit after clicking Save?

My use-case is that I am sending the edited row to api, and would like it to not close, in case there is an error message from api, but I could immediately try again (by clicking Save again).

Your help is very much appreciated.

**Comment by stsrki** [comment, marked as answer]
Yes, that is why saving events are cancellable.

One example: `RowUpdating="@((e)=>e.Cancel=true)"`

**Reply by tomironkko** [reply]
Ok, thanks, that keeps the Popup open, but how do I then conditionally close it, in case there aren't any errors?

My DataGrid attributes:

```cshtml
<DataGrid 
    TItem="Fault" 
    Data="@faults" 
    Sortable="true" 
    Editable="true"
    ShowPager="true"
    PageSize="5"
    Filterable="false"
    Hoverable="true"
    CustomFilter="@OnCustomFilter"
    RowUpdating="@((e)=>e.Cancel=true)"
    RowUpdated="@OnRowUpdated"
    EditMode= "DataGridEditMode.Popup"
    >
```

**Reply by stsrki** [reply]
That was just an example. You need to add a statement

```
async Task OnRowUpdating( CancellableRowChange<Employee, Dictionary<string, object>> e )
{
    var result = await service.TrySave( ... );

    if ( !result )
    {
        e.Cancel = true;
    }
}
```
